plugins/woocommerce/src/Blocks/BlockTypesController.php

```diff
@@ -667,4 +668,19 @@ static function ( $html ) use ( $style_handlers ) {
 
 		return $args;
 	}
+
+	/**
+	 * Change the prefered taxonomy for the breadcrumbs block on the product post type.
+	 *
+	 * @param array  $settings The settings for the breadcrumbs block.
+	 * @param string $post_type The post type.
+	 * @return array The settings for the breadcrumbs block.
+	 */
+	public function set_product_breadcrumbs_prefered_taxonomy( $settings, $post_type ) {
+		if ( 'product' === $post_type ) {
+			$settings['taxonomy'] = 'product_cat';
+		}
+
+		return $settings;
+	}
```

Comment:
1. Typo fixed in b00dd2a4b7e81d61ab4f9666bd76c2767e0f35ce.
2. I marked the method as `@internal`, so we shouldn't need `@since` anymore.
3. Added type validation in 2e906b1c72dba20b8e725e935410643acd0ee569.
4. I don't think we need to use a constant for the post type, `product` is used as a string in many other places of the codebase.